### ISSUE TITLE
🎉 Release for Sprints 239 through 240 (v3.12.0) 🎉 

### ISSUE THREAD

**Issue by rm3l** [redhat-developer/odo#6878]
/kind task
/area release-eng

Issue to track work for publishing a new release of `odo` at the end of Sprint 240.

- Target release process start date: **Jun 27, 2023**
- Errata date (when we expect the binaries to be published): **Jun 29, 2023**
- Version: **v3.12.0**
- OCP Version: **4.13**
- Once release is done, make sure link to download `latest` points to this new version: https://developers.redhat.com/content-gateway/rest/mirror/pub/openshift-v4/clients/odo/latest/

Notes:
- A draft PR will be generated by the @github-actions bot when a new GitHub Release is created (it can be a pre-release). This PR will contain all the necessary changes (`build/VERSION`, blog post, release notes, ...).
  - Feel free to review and adjust the content
  - Merge this PR once the release binaries are available on the Content Gateway

**TODO**
- ~[ ] Update the release process to skip the need to create a Bugzilla; this might require updating the errata.py script; check the release checklist. See CWFCONF-4565~ (moved to the next release)
- [x] https://github.com/redhat-developer/odo/issues/6864

**Comment by rm3l** [comment]
/status blocked

The release got delayed because we were waiting for errata docs approval, which we managed to get yesterday (Jul 4, 2023).

However, the release has been put on hold by the release team, because they are working on a critical incident (reported in RHELBLD-13214).
They will be able to resume when this incident is resolved.
No ETA is known at this time.

**Comment by rm3l** [comment]
/remove-status blocked

Binaries for v3.12.0 seem to be available now on the Content Gateway.
